CDCgov/prime-reportstream
# Experience/6309: return quality filter results from the /validate endpoint

This PR adds quality filter results to the `/validate` endpoint, without reading from or writing to the database. Quality filter results are returned in the same shape and fashion as the `/waters` endpoint- within the `destinations` portion of the JSON response.

Test Steps:
1. In PostMan, curl, etc, do a POST to the `/validate` endpoint with a file that exercises quality filters. (for example, the `simple_report_example.csv` in the repo at /prime-router/examples/submission). Observe the contents of the `destinations` array
2. Perform the same POST against the `/waters` endpoint with the same file. Observe that the contents of the `destinations` array match those in the previous step

### Testing

## Linked Issues
- Addresses #6309
